apps/minifront/src/components/staking/account/use-staking-tokens-and-filter.ts

```diff
@@ -0,0 +1,91 @@
+import { useMemo } from 'react';
+import {
+  BalancesByAccount,
+  balancesByAccountSelector,
+  useBalancesResponses,
+  useStakingTokenMetadata,
+} from '../../../state/shared';
+import {
+  Metadata,
+  ValueView,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { getDisplayDenomFromView } from '@penumbra-zone/getters/value-view';
+import { assetPatterns } from '@penumbra-zone/types/assets';
+
+/**
+ * Function to use with `reduce()` over an array of `BalancesByAccount` objects.
+ * Returns a map of accounts to `ValueView`s of the staking token.
+ */
+const toStakingTokensByAccount = (
```

Comment:
This reducer and the one below it are moved over, as-is, [from the `staking` slice](https://github.com/penumbra-zone/web/pull/1339/files#diff-7c5ec26a35f650875c894552352cecdc5e1042768e81d382bffa45e0a4dc0c1e)